### PR TITLE
jest-dom dependency sync and partial code clean up

### LINKED ISSUE #6491: Triage frontend act warnings
### Background
As detailed in [several](https://github.com/CDCgov/prime-simplereport/pull/5950) [previous](https://github.com/CDCgov/prime-simplereport/pull/3028) PR's, our frontend tests sometime get cluttered with `act` warnings. These are coming from component side effects triggering after a testing block and may contribute to flakiness / indeterminate tests. See the writeup in [this PR](https://github.com/CDCgov/prime-simplereport/pull/6439) for more details.

We're getting act warnings in places we shouldn't be though and are currently getting around it by wrapping our tests in `act`, which isn't recommended. Fixing the issue has been a bit of a headache but might be [related to a mismatched subdependency](https://github.com/testing-library/user-event/issues/1104#issuecomment-1460005394) shared by several testing library. Bob has explored this fix initially on a [separate branch](https://github.com/CDCgov/prime-simplereport/pull/6461/files) and found promising results. 

### Action requested
Triage fixing the frontend act warnings. Push to a separate feature branch any quick straightforward fixes and make tickets for the rest.

### Acceptance Criteria
- Feature branch created with easy fixes
- tickets created for remainder of work

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #6491  

## Changes Proposed
Update the dependecies that use jest-dom (jest and storybook).
Implement the new way to use userEvent in the test files that broke.
Small clean up of the act wrapping. We will need a follow up PR to finish the cleaning.

## Additional Information
We will start seeing warning asking us to remove all the act wraps T_T.

## Testing
All frontend unit testing passes.

### CODE REVIEW

File: frontend/src/app/Settings/ManageSelfRegistrationLinks.test.tsx [hunk 1]
```diff
@@ -38,42 +38,70 @@ const expectedFacilitySlug =
...
+        <ManageSelfRegistrationLinksContainer />
+      </MockedProvider>
+    ),
+  });
+
+  beforeEach(() => {
     process.env.REACT_APP_BASE_URL = testBaseUrl;
-    Object.assign(navigator, {
+
+    /*Object.assign(navigator, {
```
Comment (emyl3) [inline frontend/src/app/Settings/ManageSelfRegistrationLinks.test.tsx:57]: can we remove this comment? 👀

**Review by emyl3 — DISMISSED** [review]
LGTM! Thank you for modeling how to make use of the `userEvent` with these changes! Also ran the tests locally and saw no issues (except seeing the act warnings as you shared 😉) Local smoke test of storybook looked good too!

Thank you, @fzhao99 and @johanna-skylight, for your investigation and first pass through on this.

File: frontend/src/app/Settings/ManageSelfRegistrationLinks.test.tsx (hunk 1 above)
Comment (johanna-skylight) [reply frontend/src/app/Settings/ManageSelfRegistrationLinks.test.tsx]: thanks for catching this!

**Review by emyl3 — APPROVED** [review]

**Review by johanna-skylight — APPROVED** [review]
Approving on @fzhao99 's behalf

**Pull request merged**